docs/API/v3/service-index.md

```diff
@@ -0,0 +1,114 @@
+properties: `version` (the schema version of the service index) and `resources`  (the endpoints or capabilities of the
+package source).
+
+nuget.org's service index is located here:
+```
+https://api.nuget.org/v3/index.json
+```
+
+The service index for nuget.org looks something like is this:
+
+[!code-JSON [service-index.json](./_data/service-index.json)]
+
+## Versioning
+
+The `version` value is a SemVer 2.0.0 parseable version string which indicates the schema version of the service index.
+The V3 API mandates that the version string has a major version number of `3`. As non-breaking changes are made to the
+service index schema, the version string's minor version will be increased.
+
+Each resource in the service index is versioned independently from the service index schema version.
+
+The current schema version is `3.0.0-beta.1`.
+
+## Resources
+
+The `resources` property contains an array of resources supported by this package source.
+
+### Resource
+
+A resource is an object in the `resources` array. It represents a versioned capability of a V3 package source. A
+resource has the following properties:
+
+Name          | Type   | Required | Notes
+------------- | ------ | -------- | -----
+@id           | string | yes      | The URL to the resource
+@type         | string | yes      | A string constant representing the resource type
+clientVersion | string | no       | The client version that this resource applies to
+comment       | string | no       | A human readable description of the resource
+
+The `@id` is a URL that must be absolute and must either have the HTTP or HTTPS schema.
+
+The `@type` is used to identify the specific protocol to use when interacting with resource. The type of the resource
+is an opaque string but generally has the format:
+
+```
+{RESOURCE_NAME}/{RESOURCE_VERSION}
+```
+
+Clients are expected to hard code the `@type` values that they understand and look them up in a V3 package source's
+service index.
+
+For the sake of this V3 documentation, the documentation about different resources will essentially be grouped by the
+`{RESOURCE_NAME}` found in the service index which is analogous to grouping by scenario.
+
+There is no requirement that each resource has a unique `@id` or `@type`. It is up to the client implementation to
+determine which resource to prefer over another. One possible implementation is that resources of the same or
+compatible `@type` can be used in a round-robin fashion in case of connection failure or server error.
+
+### Client version
+
+A resource `@type` can optionally have the following form:
+
+```
+{RESOURCE_NAME}/Versioned
+```
+
+When the `@type` is suffixed with the `/Versioned` string, an additional property is required on the resource:
+`clientVersion`.
+
+This value is the minimum client version compatible with this resource. The client version is SemVer 2.0.0 compliant
```

Comment:
SemVer 2.0.0 or SemVer 1.0.0? In other words, does the client support build metadata?

Reply:
Yes, this feature was introduced in NuGet 4.0.0, which can parse SemVer 2.0.0 versions.